File: vigenere.py
def shift_char(c, kshift):
    if 65 <= ord(c) <= 90:
        c = chr((ord(c) - 65 + kshift) % 26 + 65)
        return c
    elif 97 <= ord(c) <= 122:
        c = chr((ord(c) - 97 + kshift) % 26 + 97)
        return c
    return c

def encrypt_vigenere(plaintext, keyword):
    ciphertext = ''
    n = 0

    for char in plaintext:
        if not char.isalpha():
            ciphertext += shift_char(char, 0)
        else:
            kshift = ord(keyword[n].lower()) - 97
            ciphertext += shift_char(char, kshift)
            n = (n + 1) % len(keyword)

    return ciphertext

def main():
    plaintext = input("Enter plaintext: ")
    keyword = input("Enter keyword: ")

    ciphertext = encrypt_vigenere(plaintext, keyword)
    print("Ciphertext:", ciphertext)

if __name__ == "__main__":
    main()

File: vigenere.cpp
// Author: Zuhayer Tashin
// Course: CSCI-135
// Instructor: Tong Yi
// Assignment: Lab 6B
// Write a program vigenere.cpp. It should contain a function encryptVigenere implementing this cipher:
// string encryptVigenere(string plaintext, string keyword);
// You may assume that the keyword contains only lowercase alphabetic characters a - z.

#include <iostream>
#include <cctype>
using namespace std;

char shiftChar(char c, int kshift) {
    if(int(c) >= 65 && int(c) <= 90) { 
        c += kshift; 
        if(c > 90) { 
            c -= 26; 
            return c;
        }
        return c;
    } else if(int(c) >= 97 && int(c) <= 122) { 
        int a = int(c) + kshift; 
        if(a > 122) {
            a -= 26; 
            c = a;
            return c;
        }
        c = a; 
        return c;
    }
    return c;
}

string encryptVigenere(string plaintext, string keyword) {
    char c, keyc;
    int n;
    for(int i = 0; i < plaintext.length(); i++) { 
        if(n > keyword.size() - 1) { 
            n = 0;
        }
        c = plaintext[i]; 
        keyc = keyword[n]; 
        int kshift = (int(keyc) - 97); 
        if(isalpha(c) == 0) {  
            plaintext[i] = shiftChar(c, 0);
        } else {
            plaintext[i] = shiftChar(c, kshift); 
            n++; 
        }
    }
    return plaintext;
}

int main() {
    string plaintext, kshift;

    cout << "Enter plaintext: ";
    getline(cin, plaintext);

    cout << "Enter keyword: ";
    cin >> kshift;

    cout << "Ciphertext: " << encryptVigenere(plaintext, kshift);

    return 0;
}